src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -131,34 +132,75 @@ private function createRelationsForItem($item, $formattedRelations)
                 case 'BelongsToMany':
                 case 'MorphToMany':
                     $values = $relationDetails['values'][$relationMethod] ?? [];
-                    $values = is_string($values) ? json_decode($values, true) : $values;
+                    $values = is_string($values) ? (json_decode($values, true) ?? []) : $values;
+                    $field = $relationDetails['crudFields'][0] ?? [];
 
-                    // disabling ConvertEmptyStringsToNull middleware may return null from json_decode() if an empty string is used.
-                    // we need to make sure no null value can go foward so we reassure that values is not null after json_decode()
-                    $values = $values ?? [];
+                    // if the values are multidimensional, we have additional pivot data.
+                    if (is_array($values) && is_multidimensional_array($values)) {
+                        // if the field allow duplicated pivots, we can't use sync or attach from laravel, we need to manually handle the pivot data.
+                        if ($field['allow_duplicate_pivots'] ?? false) {
+                            $keyName = $field['pivot_key_name'] ?? 'id';
```

Comment:
We could try to determine what pivot key is from either the model or database schema. Also another concern is what if its compound key ( multiple column primary key)

Reply:
We probably shouldn't do that, 99% of times it will be called `id`, and we can just ask developer to provide a different name in the rest of 1% scenarios. Less code to maintain, less possibilities of introducing bugs, less "guessing" .. let's try to KISS.

Composite keys are already what a pivot table is. Theoretically you shouldn't need an `id` column in a pivot table, unless you want to store duplicated pivots.

I would say, nothing to do here, but if you think I am missing something, please shout it out 👍